commit 574f47049e6b657202dc977e3a1c9475d6505496
Author: Bertho Stultiens <bertho@vagrearg.org>
Date:   Thu Jan 9 12:07:11 2025 +0100

    Necessary include for gettimeofday() in debian:sid.

Index: linuxcnc-2.9.3/src/emc/rs274ngc/gcodemodule.cc
===================================================================
--- a/src/emc/rs274ngc/gcodemodule.cc
+++ b/src/emc/rs274ngc/gcodemodule.cc
@@ -16,6 +16,8 @@
 //    along with this program; if not, write to the Free Software
 //    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
+#include <sys/time.h>
+
 #include <Python.h>
 #include <structmember.h>
 
